build_20230326/script.js

```diff
@@ -0,0 +1,677 @@
+		let day = lastUpdate % 100;
+        $("#notes").html(data).find("#latest").text("Latest update: "+day+" "+month+" "+year);
+    });
+    
+    $.get("../versions.html?"+lastUpdate, function(data) {
+        $("#versions-text").html(data);
+    });
+}
+
+// TODO: Use destructuring of localStorage object to tidy this up
+function loadLocal() {
+    if(localStorage.th_cj) {
+        showval = localStorage.th_cj;
+    }
+    
+    if(localStorage.th_cj_mode) {
+        activemode = localStorage.th_cj_mode;
+    }
+    
+    if(localStorage.th_cj_theme) {
+        activetheme = localStorage.th_cj_theme;
+    }
+    
+    if(localStorage.th_cj_css) {
+        showcss = localStorage.th_cj_css;
+    }
+    
+    if(localStorage.th_cj_text) {
+        showtext = localStorage.th_cj_text;
+    }
+    
+    if(localStorage.th_cj_importedmeta) {
+       importedmeta = localStorage.th_cj_importedmeta;
+    }
+    
+    if(localStorage.th_cj_importedcode) {
+       importedcode = localStorage.th_cj_importedcode;
+    }
+    
+    if(localStorage.th_cj_vertical) {
+        $("#vertical").prop("checked", localStorage.th_cj_vertical == "true");
+        setVerticalLayout();
+    }
+    
+    if(localStorage.th_cj_htmlpanel) {
+        $("#html-panel").prop("checked", localStorage.th_cj_htmlpanel == "true");
+    }
+    
+    if(localStorage.th_cj_csspanel) {
+        $("#css-panel").prop("checked", localStorage.th_cj_csspanel == "true");
+    }
+    
+    if(localStorage.th_cj_textpanel) {
+        $("#text-panel").prop("checked", localStorage.th_cj_textpanel == "true");
+    }
+    
+    if(localStorage.th_cj_bigtext) {
+        if(localStorage.th_cj_bigtext == "true") $(".editor-panel").addClass("big-text");
+        $("#big-text").prop("checked", localStorage.th_cj_bigtext == "true");
+    }
+    
+    if(localStorage.th_cj_auto) {
+        $("#auto").prop("checked", localStorage.th_cj_auto == "true");
+    }
+    
+    if(localStorage.th_cj_lastUpdate != lastUpdate && location.pathname.indexOf("/unstable") == -1) {
+        $("#info").removeClass("d-none");
+        $("#info-back").removeClass("d-none");
+        localStorage.th_cj_lastUpdate = lastUpdate;
+    }
+    
+    if(localStorage.th_cj_lowContrast) {
+        $("#low-contrast").prop("checked", localStorage.th_cj_lowContrast == "true");
+    }
+}
+
+function initEditors() {
+
+    editor = ace.edit("html-editor");
+    editor.session.setMode("ace/mode/html", () => {
+        AceColorPicker.load(ace, editor);
+    });
+    editor.setShowPrintMargin(false);
+    editor.setValue(showval);
+    editor.session.setUseWrapMode(true);
+    editor.session.on("change", function(){
+        htmlChanged=true;
+    });
+    editor.on("focus", function() {
+        if(editor.getValue()=="<!-- Enter HTML here... -->") editor.setValue("");
+    });
+    
+    css_editor = ace.edit("css-editor");
+    css_editor.session.setMode("ace/mode/css", () => {
+        AceColorPicker.load(ace, editor);
```

Comment:
Doesn't seem to have an effect for some reason.

Reply:
It seems like the second param was referring to `editor` when maybe it should have been`css_editor`?
```suggestion
        AceColorPicker.load(ace, css_editor);
```

Reply:
Update: I've tested this and this isn't it. It almost seems like ace color picker doesn't support two ace instances. wonder if there's a good workaround, will continue testing tomorrow.

Reply:
> It seems like the second param was referring to `editor` when maybe it should have been`css_editor`?

Right, that was it 😂

> Update: I've tested this and this isn't it. It almost seems like ace color picker doesn't support two ace instances. wonder if there's a good workaround, will continue testing tomorrow.

Hmm yes, the CSS-side worked only when it was renamed to plain `editor` and the html-side completely thrown out... Looks like it's a flaw of the plugin.
Well, if nothing comes out of this, at least it was a good attempt. I don't think I'm knowledgeable enough of javascript (yet) to quickly write my own solution for this.

Reply:
Yeah, that's a weird thing to hard code...I'll give it a shot! Thanks for all the help 🙏

Reply:
I think I've figured it out, it was a singular typo in the ace-colorpicker.js file (it was trying to access the global `editor` instead of the local `this.editor`). Changed that and now it works like a charm! I'm considering getting on the ace-colorpicker repo and making a PR just for this.